### ISSUE TITLE
Dask XGBoost with Dask Dataframes "empty DMatrix" and "regression_obj.cu:58: Label set is empty"

### ISSUE THREAD

**Issue by kylejn27** [dmlc/xgboost#5626]
I'm running into strange warnings while using `xgboost.dask.DaskXGBClassifier`. When using dask dataframes as inputs to the `.fit()` method I'm getting the following output:
```
task NULL connected to the tracker
task NULL connected to the tracker
task NULL connected to the tracker
task NULL connected to the tracker
task NULL got new rank 0
task NULL got new rank 1
task NULL got new rank 2
task NULL got new rank 3
WARNING:root:worker tcp://127.0.0.1:62010 has an empty DMatrix.  All workers associated with this DMatrix: {'tcp://127.0.0.1:62009', 'tcp://127.0.0.1:62008'}
WARNING:root:worker tcp://127.0.0.1:62011 has an empty DMatrix.  All workers associated with this DMatrix: {'tcp://127.0.0.1:62008', 'tcp://127.0.0.1:62009'}
WARNING:root:worker tcp://127.0.0.1:62010 has an empty DMatrix.  All workers associated with this DMatrix: {'tcp://127.0.0.1:62009', 'tcp://127.0.0.1:62008'}
WARNING:root:worker tcp://127.0.0.1:62011 has an empty DMatrix.  All workers associated with this DMatrix: {'tcp://127.0.0.1:62008', 'tcp://127.0.0.1:62009'}

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/gbm/gbtree.cc:128: Tree method is automatically selected to be 'approx' for distributed training.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/gbm/gbtree.cc:128: Tree method is automatically selected to be 'approx' for distributed training.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/gbm/gbtree.cc:128: Tree method is automatically selected to be 'approx' for distributed training.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/gbm/gbtree.cc:128: Tree method is automatically selected to be 'approx' for distributed training.

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/metric/elementwise_metric.cu:330: label set is empty
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/metric/elementwise_metric.cu:330: label set is empty

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/metric/elementwise_metric.cu:330: label set is empty
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/metric/elementwise_metric.cu:330: label set is empty

[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.
[11:42:25] WARNING: /usr/local/miniconda/conda-bld/xgboost_1584539872846/work/src/objective/regression_obj.cu:58: Label set is empty.

# these warnings keeps repeating
```

Here's example code that gets this to reproduce for me. 

```
from dask.distributed import Client, LocalCluster
import xgboost as xgb
import dask.dataframe as dd
from sklearn.datasets import make_classification
import pandas as pd

client = Client(LocalCluster())

df = make_classification(
    n_samples=10000,
    random_state=123,
)

X_array, y_array = df[0], df[1]

# merge into a dataframe and name columns
df = pd.concat([pd.Series(y_array), pd.DataFrame(X_array)], axis=1)
columns = ["target"] + ["var" + str(i) for i in range(np.shape(X_array)[1])]
df.columns = columns

# create dask dataframe
df = dd.from_pandas(df, npartitions=2)

X = df[df.columns[1:]]
y = df[df.columns[0]]

estimator = xgb.dask.DaskXGBClassifier(
    random_state=253,
    early_stopping_rounds=4,
    eval_metric=["logloss"],
    verbose_eval=False,
)
estimator.fit(X, y, eval_set=[(X,y)])
```

Here's some useful environment information:
```
Python 3.7.6
dask                      2.14.0                     py_0    conda-forge
dask-core                 2.14.0                     py_0    conda-forge
dask-glm                  0.2.0                      py_1    conda-forge
dask-ml                   1.2.0                      py_0    conda-forge
_py-xgboost-mutex         2.0                       cpu_0    conda-forge
libxgboost                1.0.2                h4a8c4bd_0    conda-forge
py-xgboost                1.0.2            py37hc8dfbb8_0    conda-forge
xgboost                   1.0.2            py37h570ac47_0    conda-forge
pandas                    1.0.3            py37h94625e5_1    conda-forge
scikit-learn              0.22.2.post1     py37h3dc85bc_0    conda-forge
```

It very well could be that I'm using XGBoost incorrectly. Though I question why I'm getting warnings from CUDA files when I'm not using my GPU for this. 

I will note that when I switch to using Dask arrays the warnings disappear

**Comment by trivialfis** [comment]
The warning means there's at least one worker was starving without any input data, which means the dataset is not balanced.  As XGBoost doesn't move data, so it raises a warning.

**Comment by kylejn27** [comment]
I see, when I set `client = Client(LocalCluster(n_workers=2))` where 2 is the number of partitions on my dataframe, the warnings go away. This makes sense.

Thanks!
